http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/CorsSupport.scala

```diff
@@ -23,20 +27,49 @@ class CorsSupport(contentSecurityPolicy: ContentSecurityPolicyConfig) {
     )
   }
 
+  /** Whether to allow any Origin header value or to restrict to the allowlist.
+   * Enable strict mode by either removing the wildcard (*) from
+   * refererConfig.validHosts or setting refererConfig.originStrict to true.  */
+  private val isOriginStrict: Boolean = !refererConfig.validHosts.contains("*") || refererConfig.originStrict
+
+  private val handleOrigin: Directive0 =
+    if (!refererConfig.enabled || !isOriginStrict) pass
+    else
+      optionalHeaderValueByType(Origin) flatMap {
+        case Some(origin) =>
+          logger.info(s"CorsSupport | Request origin detected: ${origin}. Checking validity")
+          checkSameOrigin(getValidOriginRange)
+        case None =>
+          logger.info(s"CorsSupport | Request origin not provided. Permitting")
```

Comment:
Have you checked with AppSec that this was okay?

Reply:
Will confirm today.